src/main/java/logic/modulelogic/LessonsGenerator.java

```diff
@@ -0,0 +1,131 @@
+package logic.modulelogic;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+import common.LessonType;
+import exception.InvalidUrlException;
+import exception.UnformattedModuleException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+
+import static common.Messages.MESSAGE_RETURN_SUCCESS;
+
+/**
+ * This class retrieves information from both TimetableParser and ModuleHandler classes to generate a data structure
+ * containing all the modules a user is taking and is to be used by the logic.schedulelogic component.
+ * The data structure used is an ArrayList of String[] called myLessonDetails and
+ * it contains an ArrayList of lessons in the form: startTime, endTime, day, weeks(delimited by ':').
+ */
+public class LessonsGenerator {
+    private TimetableParser myTimetableParser;
+    private ArrayList<String[]> myLessonsDetails;
+
+    public LessonsGenerator(String nusmodsLink) throws InvalidUrlException {
+        this.myTimetableParser = new TimetableParser(nusmodsLink);
+        this.myLessonsDetails = new ArrayList<>();
+    }
+
+
+    //static main method for easy in-class behaviour testing
+    public static void main(String[] args) throws InvalidUrlException, IOException, UnformattedModuleException {
+        //observe behaviour by substituting field in BackendAPI.LessonsGenerator() with other NUSMODS link
+        LessonsGenerator mylesson = new LessonsGenerator("https://nusmods.com/timetable/sem-1/share?CG1111=TUT:04,LAB:02");
+        mylesson.generate();
+        ArrayList<String[]> myLessonDetails = mylesson.getLessonDetails();
+        for (int i = 0; i < myLessonDetails.size(); i++) {
+            for (int j = 0; j < myLessonDetails.get(i).length; j++) {
+                System.out.print(myLessonDetails.get(i)[j] + " ");
+            }
+            System.out.print("\n");
+        }
+    }
+
+
+    /**
+     * Generates user's blocked time-slots based on his/her Nusmods timetable.
+     */
+    public String generate() {
+        String message;
+        message = myTimetableParser.parse();
+        if (!message.equals(MESSAGE_RETURN_SUCCESS)) {
+            return message;
+        }
+        //Key-value pair: Key = module code, Value = LessonType:Class number(delimited by :)
+        Map<String, ArrayList<String>> userLessons = myTimetableParser.getModulesMap();
+        ArrayList<String> userModules = myTimetableParser.getModulesArr();
+        // Zero based indexing: Semester 1 = 0, Semester 2 = 1
+        Integer semester = Integer.parseInt(myTimetableParser.getSemester()) - 1;
+        for (String module : userModules) {
+            ModuleHandler myModuleHandler = new ModuleHandler(module);
+            message = myModuleHandler.generateModule();
```

Comment:
![image](https://user-images.githubusercontent.com/59821379/78002050-3075ac80-7369-11ea-8dcb-68acfd0003af.png)

Should this be generateModule() and not generate(Module) in the sequence diagram